pkg/pgmodel/ingestor/copier.go

```diff
@@ -161,6 +168,9 @@ func tryRecovery(conn pgxconn.PgxConn, err error, req copyRequest) error {
 
 	// If the error was that the table is already compressed, decompress and try again.
 	if strings.Contains(pgErr.Message, "insert/update/delete not permitted") {
+		if !decompressChunk {
+			return ErrCompressedChunk
```

Comment:
hmm feels a bit weird to return the error just to catch it one step up and interpret decompressChunk in multiple places. Why don't we instead change tryRecovery to return (retryInsert bool, err error) and return false, nil here?